nestkernel/connection_creator_impl.h

```diff
@@ -617,8 +617,8 @@ ConnectionCreator::fixed_indegree_( Layer< D >& source,
           ( *positions )[ random_id ].first.get_vector( source_pos_vector );
           for ( size_t indx = 0; indx < synapse_model_.size(); ++indx )
           {
-            const double w = weight_[ indx ]->value( rng, source_pos_vector, target_pos_vector, source );
-            const double d = delay_[ indx ]->value( rng, source_pos_vector, target_pos_vector, source );
+            const double w = weight_[ indx ]->value( rng, source_pos_vector, target_pos_vector, source, tgt );
+            const double d = delay_[ indx ]->value( rng, source_pos_vector, target_pos_vector, source, tgt );
```

Comment:
Some `Parameter::value()` functions need the thread of the target node to know which internal distribution to use for random number generation. This is because distributions have a stored state, so there must be one distribution per thread. See for example https://github.com/nest/nest-simulator/blob/3dc3d38bb459895b49e44eaf258263d99a100680/nestkernel/parameter.cpp#L103-L107

Reply:
Yes, I know, I was just confused about why we only did this for the spatial connections, but I guess we never set weight and delay directly from the builders in `conn_builder` and that is why. Sorry about that :)

Reply:
In `conn_builder` the parameters are wrapped as a `conn_parameter`:

https://github.com/nest/nest-simulator/blob/d3a01157421039113b24759cce81b25a6ff1e532/nestkernel/conn_parameter.cpp#L82-L86

And we send the target node there too:
https://github.com/nest/nest-simulator/blob/d3a01157421039113b24759cce81b25a6ff1e532/nestkernel/conn_builder.cpp#L359-L360